src/app/loadout-builder/process/useProcess.ts

```diff
@@ -222,20 +228,22 @@ const groupComparator = chainComparator(
 );
 
 /**
- * This groups items for process depending on whether any general, other or raid mods are locked as follows
- * - If there are general, other or raid mods locked it groups items by (stats, masterworked, modSlot, energyType).
- * - If there are only general mods locked it groupes items by (stats, masterwork, energyType)
- * - If no general, other or raid mods are locked it groups by (stats, masterworked).
- *
- * Note that assumedMasterwork effects this.
+ * This groups items for process so we can take the first item of the group for processing.
```

Comment:
I don't understand what it means to "take the first item of the group for processing"? I actually don't understand how these groupings are used at all.

Reply:
So all things in a "group" are considered equal in regards to the LO process. So if we have a group with more than 1 item we only need to send the first one to the web worker for processing. Other items in the group will be available to the user via the swap button. Some example
- Class items, if no mods are locked then we only care if they are master worked or not (with armour upgrades considered). So that would create two groups, and we would only send two class items to process.
- Two items with identical stats, energy values, energy types, and raid sockets. These will always be identical so we would only ever send one to process.

I will try and reflect this better in the comment.

Reply:
@bhollis try that now